### PR TITLE
Draft: Move to Quarkus 2.0 version

### PR DESCRIPTION
**Pull request by kshpak**
Resolved:
1. Redirects to `/q` are not happening. Need to specify full path
2. TestSecureController: 
-- Replace deprecated method
-- Add first and the last line of PEM file to Buffer (required by libraries)
3. Jaeger config props disabled because of https://github.com/quarkusio/quarkus/issues/18131 

TODO: 
1. test with 2.0.0.Final and universe-bom
2. watch https://github.com/quarkusio/quarkus/issues/18105 (fails due to not whitelisted lines, can be suppressed in test suite)

### CODE REVIEW

File: app-full-microprofile/src/main/java/com/example/quarkus/secure/TestSecureController.java [hunk 1]
```diff
@@ -49,7 +49,7 @@ private static String generateJWT(String key) {
         JWTAuth provider = JWTAuth.create(null, new JWTAuthOptions()
                 .addPubSecKey(new PubSecKeyOptions()
                         .setAlgorithm("RS256")
-                        .setSecretKey(key)));
+                        .setBuffer(key))); // Set up private key
```
Comment (rsvoboda) [inline app-full-microprofile/src/main/java/com/example/quarkus/secure/TestSecureController.java:52]: Method name feels weird, checked with https://vertx.io/docs/vertx-auth-jwt/java/ and that's the way.

Is there any migration document for jump from Vert.x 3.9 to Vert.x 4.1?
It should be at least linked in Release notes for RHBQ 2.x

**Review by rsvoboda — COMMENTED** [review]
2.0.0.Final quarkus-bom is available

quarkus-universe-bom might be available for 2.0.0.Final as transition for users, but going forward it won't be available.

Maybe as part of move to 2.1 we should fully move to just quarkus-bom, depending what is going on with the BOM split.

File: app-full-microprofile/src/main/java/com/example/quarkus/secure/TestSecureController.java (hunk 1 above)
Comment (kshpak) [reply app-full-microprofile/src/main/java/com/example/quarkus/secure/TestSecureController.java:52]: Migration guide mentioning a link to Vertx Migration Guide. Method is recommended from class source comments https://github.com/vert-x3/vertx-auth/blob/master/vertx-auth-common/src/main/java/io/vertx/ext/auth/PubSecKeyOptions.java

**Comment by rsvoboda** [comment]
> TODO: test with 2.0.0.Final and universe-bom

What's the status here? 
I think for 2.1/2.2 we will need to rethink quarkus-universe-bom related tests as there will be other way to compose the platform, one monolithic quarkus-universe-bom is going to be no longer suggested way forwarwards.

> TODO:  watch QuarkusSmallRyeTracingDynamicFeature bean is removed during build quarkusio/quarkus#18105 (fails due to not whitelisted lines, can be suppressed in test suite)

I would add whitelist item with comment `// TODO remove once https://github.com/quarkusio/quarkus/issues/18105 is resolved`

**Comment by kshpak** [comment]
1. All modules use quarkus-bom, no needed to test universe-bom
2. I will add whitelisted item with TODO
3. We need to rethink main testRuntime() method because of https://github.com/quarkusio/quarkus/issues/18527. There are few tests that trying to access `index.html`, but their generated applications do not contain META-INF folder with index.html

**Comment by rsvoboda** [comment]
for now we can have explicit dependency on "resteasy" extension in cases where index.html is not available.

Extensions with index.hml present:
```
find . | grep extension-codestarts | grep index.entry.qute.html
./independent-projects/tools/base-codestarts/target/classes/codestarts/quarkus/extension-codestarts/resteasy-codestart/base/src/main/resources/META-INF/resources/index.entry.qute.html
./independent-projects/tools/base-codestarts/target/classes/codestarts/quarkus/extension-codestarts/spring-web-codestart/base/src/main/resources/META-INF/resources/index.entry.qute.html
./independent-projects/tools/base-codestarts/target/classes/codestarts/quarkus/extension-codestarts/resteasy-reactive-codestart/base/src/main/resources/META-INF/resources/index.entry.qute.html
./independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/extension-codestarts/resteasy-codestart/base/src/main/resources/META-INF/resources/index.entry.qute.html
./independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/extension-codestarts/spring-web-codestart/base/src/main/resources/META-INF/resources/index.entry.qute.html
./independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/extension-codestarts/resteasy-reactive-codestart/base/src/main/resources/META-INF/resources/index.entry.qute.html
./devtools/project-core-extension-codestarts/target/classes/codestarts/quarkus/extension-codestarts/funqy-http-codestart/base/src/main/resources/META-INF/resources/index.entry.qute.html
./devtools/project-core-extension-codestarts/target/classes/codestarts/quarkus/extension-codestarts/resteasy-qute-codestart/base/src/main/resources/META-INF/resources/index.entry.qute.html
./devtools/project-core-extension-codestarts/target/classes/codestarts/quarkus/examples/undertow-websockets-example/base/src/main/resources/META-INF/resources/index.entry.qute.html
./devtools/project-core-extension-codestarts/src/main/resources/codestarts/quarkus/extension-codestarts/funqy-http-codestart/base/src/main/resources/META-INF/resources/index.entry.qute.html
./devtools/project-core-extension-codestarts/src/main/resources/codestarts/quarkus/extension-codestarts/resteasy-qute-codestart/base/src/main/resources/META-INF/resources/index.entry.qute.html
./devtools/project-core-extension-codestarts/src/main/resources/codestarts/quarkus/examples/undertow-websockets-example/base/src/main/resources/META-INF/resources/index.entry.qute.html
```

**Comment by kshpak** [comment]
Update (still draft):
logging-json removed from supportedExtensionsWithCodestart (no codestart)
config-yaml commented out as it doesn't produce `index.html` (we should decide if it can be removed from list)
supportedExtensionsSubsetC requires `quarkus-resteasy` in order to have `index.html`
two whilelist patterns added (please check respective github issues)

@jsmrcka please check ArtifactGeneratorTest with your instrumentation code (2.0.2.Final does not produce any more SpringGreetingController.java)

File: testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusTest.java [hunk 2]
```diff
@@ -159,7 +160,7 @@ public void testRuntime(TestInfo testInfo, List<CodeQuarkusExtensions> extension
             pA = runCommand(cmd, appDir, runLogA);
             
             // It takes time to download the Internet
-            long timeoutS = 10 * 60;
+            long timeoutS = 1 * 60;
```
Comment (rsvoboda) [inline testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusTest.java:163]: needs to be reverted

**Pull request closed without merging**